### PR TITLE
Enable HPOS in the e2e environment using env var

### PR DESCRIPTION
**Pull request by MrJnrman**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR updates the necessary scripts and workflow files to enable HPOS using an env var instead of passing the `--cot` flag. The reasoning here is this the `test-env-setup.sh` is used elsewhere and the use of env vars allows more efficient automation.

#### Set up the local environment with COT enabled using wp-env

1. Run `nvm use` into your terminal
2. Run `pnpm install`
3. Run  `pnpm build`
4. Run `pnpm run env:test:cot --filter=woocommerce`
5. Once the environment is ready, head to http://localhost:8086/wp-admin/plugins.php and check that the `WooCommerce Enable COT` plugin is installed and activated.
6. Go to `WooCommerce > Settings > Advanced` and check there is a new tab labelled `Custom data stores` that looks like the following:
image

#### Set up the local environment with COT disabled using wp-env

1. Run `nvm use` into your terminal
2. Run `pnpm install`
3. Run  `pnpm build`
4. Run `pnpm run env:test --filter=woocommerce`
5. Once the environment is ready, head to http://localhost:8086/wp-admin/plugins.php and check that the `WooCommerce Enable COT` is missing.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: .github/workflows/pr-build-and-e2e-tests.yml [hunk 1]
```diff
@@ -127,6 +131,8 @@ jobs:
 
             - name: Load docker images and start containers.
               working-directory: plugins/woocommerce
+              env:
+                ENABLE_HPOS: 1
```
Comment (tammullen) [inline .github/workflows/pr-build-and-e2e-tests.yml]: Think this is a mistake to enable only for performance in this regular PR workflow?

**Review by tammullen — COMMENTED** [review]
FWIW the k6 tests can run against sites using Order Tables using this config variable which means the requests use the correct HPOS URLs https://github.com/woocommerce/woocommerce/blob/cfe8f76be95ed7adc976cb830fd3e13b5a40b77a/plugins/woocommerce/tests/performance/config.js#L14

```
env:
                  COT: true
```

But maybe that is something for a separate PR

File: .github/workflows/pr-build-and-e2e-tests.yml (hunk 1 above)
Comment (MrJnrman) [reply .github/workflows/pr-build-and-e2e-tests.yml]: That's right! Thanks for catching that

**Comment by lanej0** [comment]
@MrJnrman Just got a merge conflict on this one now.  Once you've got that resolved, I'll give it a final 👀 and 👍🏼

**Review by lanej0 — APPROVED** [review]
Checks out and works as described. 👍🏼

**Pull request merged**